components/locale_core/src/preferences/extensions/unicode/keywords/calendar.rs

```diff
@@ -56,7 +58,8 @@ enum_keyword!(
              ("umalqura" => Umalqura),
              ("tbla" => Tbla),
              ("civil" => Civil),
-             ("rgsa" => Rgsa)
+             ("rgsa" => Rgsa),
+             ("iran" => HijriIran)
```

Comment:
We can't do this until we have a BCP47 id for this, I think?

Reply:
Yeah, for it to be added here we should upstream it first. You can implement a calendar though without the bcp47.

Reply:
Done, just had to make calendar_algorithm to return None for this, which is also done in  https://github.com/unicode-org/icu4x/blob/c6985bcbdcd72d2f087ebef77068a0000766ff3d/components/calendar/src/cal/julian.rs#L202-L204 Julian and Iso's implementations also.